pkg/controllers/gateway/gatewayclass_controller.go

```diff
@@ -46,9 +57,67 @@ type GatewayClassReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.1/pkg/reconcile
 func (r *GatewayClassReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := ctrllog.FromContext(ctx)
+
+	previous := &gatewayv1beta1.GatewayClass{}
+	err := r.Client.Get(ctx, client.ObjectKey{Namespace: req.Namespace, Name: req.Name}, previous)
+	if err != nil {
+		if err := client.IgnoreNotFound(err); err == nil {
+			return ctrl.Result{}, nil
+		} else {
+			log.Error(err, "Unable to fetch GatewayClass")
+			return ctrl.Result{}, err
+		}
+	}
 
-	// TODO(user): your logic here
+	if previous.Spec.ControllerName != ControllerName {
+		log.Info("Not a GatewayClass for this controller", "ControllerName", previous.Spec.ControllerName)
+		return ctrl.Result{}, nil
+	}
+
+	if previous.Status.Conditions != nil &&
+		len(previous.Status.Conditions) == 1 &&
+		previous.Status.Conditions[0].Type == string(gatewayv1beta1.GatewayClassConditionStatusAccepted) &&
+		previous.Status.Conditions[0].Status == "True" {
+		log.Info("GatewayClass already Accepted")
+		return ctrl.Result{}, nil
+	}
+
+	gatewayclass := previous.DeepCopy()
+	supportedClasses := getSupportedClasses()
+	if !contains(supportedClasses, previous.Name) {
```

Comment:
```suggestion
	if !slice.ContainsString(supportedClasses, previous.Name) {
```

The [`slice`](https://github.com/Kuadrant/multi-cluster-traffic-controller/blob/main/pkg/_internal/slice/slice.go#L22) package already has a function we could use

Reply:
Handy